langchain/embeddings/google_palm.py

```diff
@@ -0,0 +1,39 @@
+"""Wrapper arround Google's PaLM Embeddings APIs."""
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, root_validator
+
+from langchain.embeddings.base import Embeddings
+from langchain.utils import get_from_dict_or_env
+
+
+class GooglePalmEmbeddings(BaseModel, Embeddings):
+    client: Any
+    google_api_key: Optional[str]
+    model_name: str = "models/embedding-gecko-001"
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate api key, python package exists."""
+        google_api_key = get_from_dict_or_env(
+            values, "google_api_key", "GOOGLE_API_KEY"
+        )
+        try:
+            import google.generativeai as genai
+            from google.generativeai import text
```

Comment:
stylistic: could we just use `genai.text`. 1 bc we already imported genai, 2 bit more readable (was confused by bare `text`)

Reply:
Turns out .generate_text and .generate_embeddings are already lifted onto the genai module anyway, so cleaned this up pretty well.